src/Content/Lock.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cms\App;
+use Kirby\Cms\User;
+use Kirby\Toolkit\Str;
+
+/**
+ * The Lock class provides information about the
+ * locking state of a content version, depending
+ * on the timestamp and locked user id
+ *
+ * @internal
+ * @since 5.0.0
+ *
+ * @package   Kirby Content
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class Lock
+{
+	public function __construct(
+		protected User $user,
+		protected int $modified,
+	) {
+	}
+
+	/**
+	 * Creates a lock for the given version by
+	 * reading the modification timestamp and
+	 * lock user id from the version.
+	 */
+	public static function for(
+		Version $version
+	): static {
+		// if the version does not exist, it cannot be locked
+		if ($version->exists() === false) {
+			// create an open lock for the current user
+			return new static(
+				user: App::instance()->user(),
+				modified: 0
+			);
+		}
+
+		// Read the locked user id from the version
+		$userId = $version->read()['lock'] ?? null;
```

Comment:
```suggestion
		$userId = $version->read('default')['lock'] ?? null;
```

I think it would be good to make it explicit here that we want the default language. Just not sure if passing as string or actual language object.

Reply:
I think this really depends on how we want to move forward. We could still think about locking individual languages. But I guess this way would be easiest for now.

Reply:
Once we support it, sure. But for now I would still make it explicit so it reads more explicitly without looking up what the default for the parameter is.